src/plugins/console/public/lib/utils/index.ts

```diff
@@ -94,3 +96,54 @@ export function splitOnUnquotedCommaSpace(s: string) {
   arr.push(buffer);
   return arr;
 }
+
+export const replaceVariables = (
+  requests: RequestArgs['requests'],
+  variables: DevToolsVariable[]
+) => {
+  const urlRegex = /(\${\w+})/g;
+  const bodyRegex = /("\${\w+}")/g;
+  const updatedRequests = requests.map((req) => {
+    if (urlRegex.test(req.url)) {
+      req.url = req.url.replaceAll(urlRegex, (match) => {
+        // Sanitize variable name
+        const key = match.replace('${', '').replace('}', '');
+        const variable = variables.find(({ name }) => name === key);
+
+        return variable?.value ?? match;
+      });
+    }
+
+    if (req.data.length) {
+      if (bodyRegex.test(req.data[0])) {
+        const data = req.data[0].replaceAll(bodyRegex, (match) => {
+          // Sanitize variable name
+          const key = match.replace('"${', '').replace('}"', '');
+          const variable = variables.find(({ name }) => name === key);
+
+          if (variable) {
+            const { value } = variable;
+            if (
```

Comment:
Can we add a comment here about the intention of this condition?

Reply:
Added